consent/handler.go

```diff
@@ -298,13 +300,21 @@ func (h *Handler) AcceptLoginRequest(w http.ResponseWriter, r *http.Request, ps
 		return
 	}
 
-	var p HandledLoginRequest
+	var (
+		p   HandledLoginRequest
+		err error
+	)
 	d := json.NewDecoder(r.Body)
 	d.DisallowUnknownFields()
-	if err := d.Decode(&p); err != nil {
+	if err = d.Decode(&p); err != nil {
 		h.r.Writer().WriteErrorCode(w, r, http.StatusBadRequest, errors.WithStack(err))
 		return
 	}
+
+	defer func(p *HandledLoginRequest, err error) {
```

Comment:
A particular reason why this is a defer?

Reply:
The alternative is to call `recordAcceptLoginRequest(h.r.PrometheusManager(), p, err)` before every return. I thought the defer would work nicely here.

Reply:
Ah I see, since you're passing the error down to prometheus you want that in there right?

Reply:
exactly! I still need to make sure that works as I expect it to; apparently the way that defer is evaluated is a little weird, which is why its an anonymous function and not just directly calling `recordConsentRequest`.